skyportal/handlers/api/tag.py

```diff
@@ -0,0 +1,238 @@
+        with self.Session() as session:
+            tag = session.scalars(
+                ObjTagOption.select(session.user_or_token).where(
+                    ObjTagOption.id == tag_id
+                )
+            ).first()
+
+            if not tag:
+                return self.error("Tag not found", status=404)
+
+            session.delete(tag)
+            session.commit()
+
+            return self.success(f"Successfully deleted tag {tag}")
+
+
+class ObjTagHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        """Get all tag-obj associations or filter by obj_id/objtagoption_id"""
+        obj_id = self.get_query_argument("obj_id", None)
+        objtagoption_id = self.get_query_argument("objtagoption_id", None)
+
+        with self.Session() as session:
+            query = ObjTag.select(session.user_or_token)
+
+            if obj_id:
+                query = query.where(ObjTag.obj_id == obj_id)
+            if objtagoption_id:
+                query = query.where(ObjTag.objtagoption_id == objtagoption_id)
+
+            associations = session.scalars(query).all()
+            return self.success(associations)
+
+    @auth_or_token
+    def post(self):
+        """Create a new tag-obj association"""
+        data = self.get_json()
+        objtagoption_id = data.get("objtagoption_id")
+        obj_id = data.get("obj_id")
+
+        if not objtagoption_id or not obj_id:
+            return self.error("Both `objtagoption_id` and `obj_id` must be provided")
+
+        with self.Session() as session:
+            # Check if association already exists
+            if session.scalars(
+                ObjTag.select(session.user_or_token)
+                .where(ObjTag.objtagoption_id == objtagoption_id)
+                .where(ObjTag.obj_id == obj_id)
+            ).first():
+                return self.error("This tag-obj association already exists")
+
+            # Verify tag exists
+            if not session.scalars(
+                ObjTagOption.select(session.user_or_token).where(
+                    ObjTagOption.id == objtagoption_id
+                )
+            ).first():
+                return self.error("Specified tag does not exist", status=404)
+
+            # Verify obj exists
+            if not session.scalars(
+                Obj.select(session.user_or_token).where(Obj.id == obj_id)
+            ).first():
+                return self.error("Specified obj does not exist", status=404)
+
+            if hasattr(self.current_user, "created_by"):
+                author_id = self.current_user.created_by.id
+            else:
+                author_id = self.current_user.id
+
+            new_assoc = ObjTag(
+                objtagoption_id=objtagoption_id, obj_id=obj_id, author_id=author_id
+            )
+            session.add(new_assoc)
+            session.commit()
+
+            return self.success(new_assoc)
+
+    @auth_or_token
+    def patch(self, association_id):
```

Comment:
so while I understand why we have a POST/PATCH/DELETE for the tag options (users want to add new tag options, delete them, or edit existing one), I really don't think we need any PATCH/PUT for the tags on an object. Since it's just about associating things anyways from the frontend there will never be a scenario where the user will want to click on a tag to replace it with another. It will always be as simple as removing an unwanted tag to then add another. So let's keep this simple, and just remove the `patch` method for the `ObjTagHandler`.

Reply:
and just to be extra clear, we do want to keep it for `ObjTagOptionHandler` though